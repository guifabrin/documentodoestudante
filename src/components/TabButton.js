import {Text, TouchableHighlight, View} from 'react-native';
import React from 'react';
import Ionicons from 'react-native-vector-icons/Ionicons';
import {COLORS} from '../config/variables';
import styles from './TabButton.styles';

const TabButton = ({handler, title, icon, active}) => {
  return (
    <TouchableHighlight
      style={styles.container}
      onPress={handler}
      underlayColor={COLORS.SECONDARY}
    >
      <View>
        <View
          style={[
            styles.activeBar,
            {backgroundColor: active ? COLORS.ACTIVE : 'transparent'},
          ]}
        />
        <View
          style={[
            styles.tab,
            {backgroundColor: active ? COLORS.SECONDARY : 'transparent'},
          ]}
        >
          <Ionicons name={icon} style={styles.icon} />
          <Text style={styles.tabTitle}>{title}</Text>
        </View>
      </View>
    </TouchableHighlight>
  );
};

export default TabButton;
