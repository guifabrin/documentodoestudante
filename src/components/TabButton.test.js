import React from 'react';
import renderer from 'react-test-renderer';
import TabButton from './TabButton';

describe('TabButton', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <TabButton
          handler={() => {}}
          title="Test Button"
          icon="ios-add"
          active={false}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders with active color when active is true', () => {
    const tree = renderer
      .create(
        <TabButton
          handler={() => {}}
          title="Test Button"
          icon="ios-add"
          active={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
