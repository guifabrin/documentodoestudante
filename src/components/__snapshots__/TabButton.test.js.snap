// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton renders correctly 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        [
          {
            "height": 5,
          },
          {
            "backgroundColor": "transparent",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-around",
            "padding": 5,
          },
          {
            "backgroundColor": "transparent",
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          {
            "color": "#ebebeb",
            "textAlign": "center",
          }
        }
      >
        Test Button
      </Text>
    </View>
  </View>
</View>
`;

exports[`TabButton renders with active color when active is true 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        [
          {
            "height": 5,
          },
          {
            "backgroundColor": "#0A3A5C",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-around",
            "padding": 5,
          },
          {
            "backgroundColor": "#32a5f9",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            {
              "color": "#ebebeb",
            },
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ï„€
      </Text>
      <Text
        style={
          {
            "color": "#ebebeb",
            "textAlign": "center",
          }
        }
      >
        Test Button
      </Text>
    </View>
  </View>
</View>
`;
